packages/flutter/lib/src/cupertino/context_menu.dart

```diff
@@ -0,0 +1,1224 @@
+  BuildContext context,
+  double scale,
+  double opacity,
+);
+
+// Given a GlobalKey, return the Rect of the corresponding RenderBox's
+// paintBounds in global coordinates.
+Rect _getRect(GlobalKey globalKey) {
+  assert(globalKey.currentContext != null);
+  final RenderBox renderBoxContainer = globalKey.currentContext.findRenderObject();
+  final Offset containerOffset = renderBoxContainer.localToGlobal(renderBoxContainer.paintBounds.topLeft);
+  return containerOffset & renderBoxContainer.paintBounds.size;
+}
+
+// The context menu arranges itself slightly differently based on the location
+// on the screen of [ContextMenu.child] before the [ContextMenu] opens.
+enum _ContextMenuLocation {
+  center,
+  left,
+  right,
+}
+
+/// A full-screen modal route that opens when the [child] is long-pressed.
+///
+/// When open, the ContextMenu shows the child, or [preview] if given, in a
+/// large full-screen Overlay with a list of buttons specified by [actions]. The
+/// child/preview is placed in an Expanded widget so that it will grow to fill
+/// the Overlay if its size is unconstrained.
+///
+/// When closed, the ContextMenu simply displays the child as if the ContextMenu
+/// were not there. Sizing and positioning is unaffected. The menu can be closed
+/// like other [PopupRoute]s, such as by tapping the background or by calling
+/// `Navigator.pop(context)`. Unlike PopupRoute, it can also be closed by
+/// swiping downwards.
+///
+/// The [preview] parameter is most commonly used to display a slight variation
+/// of [child]. See [preview] for an example of rounding the child's corners and
+/// allowing its aspect ratio to expand, similar to the iPhoto app on iOS.
+///
+/// {@tool dartpad --template=stateless_widget_material}
+///
+/// This sample shows a very simple ContextMenu for an empty red 100x100
+/// Container. Simply long press on it to open.
+///
+/// ```dart
+/// Widget build(BuildContext context) {
+///   return Scaffold(
+///     key: scaffoldKey,
+///     body: Center(
+///       child: Container(
+///         width: 100,
+///         height: 100,
+///         child: ContextMenu(
+///           child: Container(
+///             color: Colors.red,
+///           ),
+///           actions: <ContextMenuAction>[
+///             ContextMenuAction(
+///               child: const Text('Action one'),
+///               onPressed: () {
+///                 Navigator.pop(context);
+///               },
+///             ),
+///             ContextMenuAction(
+///               child: const Text('Action two'),
+///               onPressed: () {
+///                 Navigator.pop(context);
+///               },
+///             ),
+///           ],
+///         ),
+///       ),
+///     ),
+///   );
+/// }
+/// ```
+/// {@end-tool}
+///
+/// See also:
+///
+///   * [Apple's HIG for Context Menus](https://developer.apple.com/design/human-interface-guidelines/ios/controls/context-menus/)
+class ContextMenu extends StatefulWidget {
```

Comment:
Do we want the `Cupertino` prefix? I found this PR https://github.com/flutter/flutter/pull/20267 but couldn't find the rationale.

Reply:
CC @HansMuller

Reply:
I think we've established a convention already:
```
CupertinoActivityIndicator
CupertinoDatePicker
CupertinoTimerPicker
CupertinoTabBar
CupertinoTabView
CupertinoTabScaffold
CupertinoSegmentedControl<T>
CupertinoSliverRefreshControl
CupertinoButton
CupertinoUserInterfaceLevel
CupertinoPicker
CupertinoApp
CupertinoSlider
CupertinoPageScaffold
CupertinoAlertDialog
CupertinoDialog
CupertinoPopupSurface
CupertinoDialogAction
CupertinoTheme
CupertinoTextField
CupertinoNavigationBar
CupertinoSliverNavigationBar
CupertinoNavigationBarBackButton
CupertinoSwitch
CupertinoActionSheet
CupertinoActionSheetAction
CupertinoScrollbar
CupertinoTextSelectionToolbar
```

Reply:
That looks like a trend to me haha.  I'm changing it.